src/scripts/dialog.js

```diff
@@ -0,0 +1,218 @@
+    let output = '';
+    if (outputTag) {
+      output = outputTag.getHtml();
+    } else {
+      // set noOutput to be checked
+      dialog.setValueOf('tab-basic', 'no-output', true);
+    }
+
+    const codeArea = dialog.getContentElement('tab-basic', 'code').getElement();
+    codeArea.setHtml(editScriptAreaHTML(languageDictionary[language], code, output));
+
+    this.resize(500, 500);
+  },
+  onCancel() {
+    const dialog = this;
+    const editorLanguage = getLanguage(editor);
+    const dialogLanguage = dialog.getValueOf('tab-basic', 'language');
+    if (editorLanguage !== dialogLanguage) {
+      activateThebelab(thebelabConfig);
+    }
+  },
+  contents: [
+    {
+      id: 'tab-basic',
+      label: 'Basic Settings',
+      elements: [
+        {
+          type: 'select',
+          id: 'language',
+          label: 'Select language for page:',
+          items: [['Python 3'], ['Julia'], ['R'], ['Octave'], ['SageMath']],
+          default: 'Python 3',
+          onChange() {
+            const element = this.getDialog().getContentElement('tab-basic', 'code').getElement();
+            element.setHtml(editScriptAreaHTML(languageDictionary[this.getValue()]));
+            activateThebelab(thebelabConfig);
+          },
+        },
+        {
+          type: 'html',
+          id: 'code',
+          // html will be set in onShow()
+          html: '',
+        },
+        {
+          type: 'hbox',
+          id: 'checkboxes',
+          widths: ['25%', '25%', '50%'],
+          children: [
+            {
+              type: 'checkbox',
+              id: 'no-output',
+              label: 'Insert without output',
+            },
+            {
+              type: 'checkbox',
+              id: 'no-code',
+              label: 'Insert with code hidden',
+            },
+            {
+              type: 'html',
+              html: '',
+            },
+          ],
+        },
+        {
+          type: 'html',
+          id: 'insert-warning',
+          html: insertWarning(),
+        },
+      ],
+    },
+  ],
+  onOk() {
+    const dialog = this;
+    const widget = dialog.getModel(editor);
+    const language = languageDictionary[dialog.getValueOf('tab-basic', 'language')];
+
+    // changes the data-language attributes of all pre tags in editor
+    changeAllLanguages(editor, language);
+
+    const noOutput = dialog.getValueOf('tab-basic', 'no-output');
+    const noCode = dialog.getValueOf('tab-basic', 'no-code');
+    const cm = document.querySelector('.cke_dialog_contents .thebelab-input .CodeMirror').CodeMirror;
+
+    let output = document.querySelector('.cke_dialog_contents .jp-OutputArea-output');
+    // the output will contain a pre tag if run by binder
+    if (output.children.length !== 0 && output.children[0].tagName === 'PRE') {
+      output = output.children[0].innerHTML;
+    } else {
+      output = output.innerHTML;
+    }
+
+    // pass to widgets to figure out
```

Comment:
There are a lot of comments in here which are hard to decipher. It's not immediately obvious to me what some parts of the code do.

I suggest we make more comments in other parts of the code sometime later (I haven't been commenting very well myself).

Reply:
I agree it's difficult to understand. I will add more comments.